sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -541,12 +541,55 @@ def test_binomial_rewrite():
     k = Symbol('k', integer=True)
     x = Symbol('x')
 
-    assert binomial(n, k).rewrite(
-        factorial) == factorial(n)/(factorial(k)*factorial(n - k))
-    assert binomial(
-        n, k).rewrite(gamma) == gamma(n + 1)/(gamma(k + 1)*gamma(n - k + 1))
+    assert binomial(n, k).rewrite(factorial) == factorial(n)/(factorial(k)*factorial(n - k))
+    assert binomial(n, k).rewrite(gamma) == gamma(n + 1)/(gamma(k + 1)*gamma(n - k + 1))
     assert binomial(n, k).rewrite(ff) == ff(n, k) / factorial(k)
     assert binomial(n, x).rewrite(ff) == binomial(n, x)
+    assert binomial(n, k).rewrite(gamma).rewrite('tractable') == exp(-loggamma(k + 1))*exp(loggamma(n + 1))*exp(-loggamma(-k + n + 1))
+
+
+def test_multinomial():
+    from sympy import I
+    n, k = symbols('n k', integer=True)
+    z = Symbol('z', zero=True)
+    nt = Symbol('nt', integer=False)
+    kp = Symbol('kp', integer=True, positive=True)
+
+    assert multinomial(0, 0) == 1
+    assert multinomial(1, 2) == 3
+    assert multinomial(1, -2) == 0
+    assert multinomial(-1, -2) == 0
+    assert multinomial(-pi, 2*pi) == gamma(1 + pi)/(gamma(1 - pi)*gamma(1 + 2*pi))
+    assert multinomial(1, 1, 1) == 6
+    assert multinomial(S.Half, S.Half) == 4/pi
+
+    assert multinomial(k, n - k).equals(binomial(n, k))
+
+    assert multinomial(n, n).is_integer
+    assert multinomial(nt, k).is_integer is False
+
+    assert expand_func(multinomial(n, -2)) == 0
+    assert (expand_func(multinomial(n, -pi))).equals((pi - n)*gamma(n - pi)/(pi*n*gamma(-pi)*gamma(n)))
+    assert expand_func(multinomial(n, 2)) == (n + 1)*(n + 2)/2
+    assert expand_func(multinomial(n, 2, 3)) == (n + 1)*(n + 2)*(n + 3)*(n + 4)*(n + 5)/12
+    assert expand_func(multinomial(n, n - 3)) == (n - 2)*(n - 1)*gamma(2*n)/(n*(2*n - 2)*(2*n - 1)*gamma(n)**2)
+
+    assert multinomial(n, 3).expand(func=True) ==  n**3/6 + n**2 + 11*n/6 + 1
+
+    assert multinomial(kp, kp + 1).expand(func=True) == 4*kp*gamma(2*kp)/(kp**2*gamma(kp)**2 + kp*gamma(kp)**2) + 2*gamma(2*kp)/(kp**2*gamma(kp)**2 + kp*gamma(kp)**2)
+    assert expand_func(multinomial(n, z)) == (n + z)*gamma(n + z)/(n*z*gamma(n)*gamma(z))
+
+    assert multinomial(I, 2*I) == gamma(1 + 3*I)/(gamma(1 + I)*gamma(1 + 2*I))
```

Comment:
If `multinomial` can't return a numerical value then I think it should just remain unevaluated and the user can rewrite as `gamma` if desired.

Reply:
since i is a number(complex tho), multinomial(i , 2i) is also a numerical value ... it's a complex number technically, `gamma(1 + 3*I)/(gamma(1 + I)*gamma(1 + 2*I))` is a complex number approximately [`0.16529651.. + 0.4593069... i` ](http://www.wolframalpha.com/input/?i=gamma(1%20%2B%203*I)%2F(gamma(1%20%2B%20I)*gamma(1%20%2B%202*I)))
but since gamma just doesn't simplify for complex numbers we get it in terms of gamma ...
for eg
```python3
>>> gamma(I + 3)
gamma(3 + I)
```
whereas it equals [0.96 + 1.33i](http://www.wolframalpha.com/input/?i=gamma(i%20%2B%203)%20%3D%3D%20%3F)
hence ...
```python3
>>> multinomial(I, I)
gamma(1 + 2*I)/gamma(1 + I)**2
```
which is really a numerical value ... 
?

Reply:
should I raise this as another issue ... or is it intended this way ?

Reply:
I mean that there is no point in seeing the multinomial written in terms of gammas if it isn't going to evaluate.

Reply:
then that would call for reimplementation of a lot of things ... 
because
```python3
>>> I.is_number
True
```
and it is how the binomial function behaves as well 
```python3
>>> binomial(2*I,4-I)
gamma(1 + 2*I)/(gamma(-3 + 3*I)*gamma(5 - I))
```
so I don't think there'd be any harm in leaving it as it is.
wdys @smichr ?

Reply:
`binomial` shouldn't do this, either. There is nothing instructive in the rewrite. I don't know what `binomial(2, I)` is and seeing it as `2/(gamma(1 + I)*gamma(3 - I))` is not helpful. If I want to know the value I can evaluate.

note, too, that `I.is_number` and `I.is_Number` don't mean the same thing. Basically what I mean by getting a "Number" result is that `pure_complex(x, or_real=True)` should not be None.